src/special/build_runner.zig

```diff
@@ -258,13 +262,30 @@ fn processIncludeDirs(
     try include_dirs.ensureUnusedCapacity(builder.allocator, dirs.len);
 
     for (dirs) |dir| {
-        const candidate: []const u8 = switch (dir) {
-            .path => |path| copied_from_zig.getPath(path, builder) orelse continue,
-            .path_system => |path| copied_from_zig.getPath(path, builder) orelse continue,
-            else => continue,
-        };
-
-        include_dirs.putAssumeCapacity(candidate, {});
+        switch (dir) {
+            .path, .path_system => |path| {
+                const resolved_path = copied_from_zig.getPath(path, builder) orelse continue;
+                include_dirs.putAssumeCapacity(resolved_path, {});
+            },
+            .other_step => |other_step| {
+                if (other_step.generated_h) |header| {
+                    if (header.path) |path| {
+                        include_dirs.putAssumeCapacity(std.fs.path.dirname(path).?, {});
+                    }
+                }
+                if (other_step.installed_headers.items.len > 0) {
+                    const path = builder.pathJoin(&.{
+                        other_step.step.owner.install_prefix, "include",
+                    });
+                    include_dirs.putAssumeCapacity(path, {});
+                }
```

Comment:
this block has two potential calls to `include_dirs.putAssumeCapacity` but there has only been capacity reserved for one per loop iteration